### ISSUE TITLE
 eval_acc parameter

### ISSUE THREAD

**Issue by rnyak** [IndicoDataSolutions/finetune#448]
**Describe the bug**
I set the eval_acc = true, and val_size = 1000. I am fine-tuning the Classifier model for 3 epochs. I get 90% training 82% test set accuracies, but when I check the TensorBoard accuracy plot,  I see val accuracy is 49%. That does not seem correct to me. 

I am not sure if the eval_acc is calculated correctly. 

**Expected behavior**
How can we print put val accuracy during fine tuning at least for epoch?

**Comment by benleetownsend** [comment]
I will look into this now, that certainly does not seem correct. 

As for printing out the validation  accuracy, you can set the config option `debugging_logs=True` to get the full logger output including the validation set metrics.

**Comment by benleetownsend** [comment]
```
import numpy as np
from sklearn.model_selection import train_test_split

from finetune import Classifier
from finetune.datasets.stanford_sentiment_treebank import StanfordSentimentTreebank

if __name__ == "__main__":
    dataset = StanfordSentimentTreebank(nrows=1000).dataframe
    model = Classifier(
        debugging_logs=True,
        max_length=64,
        val_size=0.1,
        eval_acc=True,
        n_epochs=1,
    )
    trainX, testX, trainY, testY = train_test_split(dataset.Text.values, dataset.Target.values, test_size=0.3, random_state=42)
    model.fit(trainX, trainY)
    print(np.mean(np.array(model.predict(testX)) == testY))
```

**Comment by benleetownsend** [comment]
Just had a thought, could you try re-running with `chunk_long_sequences=False` and see what happens here. This is the only thing i can think of that might change the behaviour between validation and testing in such a bizarre way.

**Comment by rnyak** [comment]
In my runs, I am already using setting chunk_long_sequences as False. 

The final number I got from the above example is 0.6633333333333333.

and near the end of logs I got 

```INFO:tensorflow:Finished evaluation at 2019-12-04-21:34:08
INFO:tensorflow:Finished evaluation at 2019-12-04-21:34:08
INFO:tensorflow:Saving dict for global step 47: Accuracy = 0.6142857, global_step = 47, loss = 0.6300657
INFO:tensorflow:Saving dict for global step 47: Accuracy = 0.6142857, global_step = 47, loss = 0.6300657
INFO:tensorflow:Loss for final step: 0.5167145.
INFO:tensorflow:Loss for final step: 0.5167145.```

**Comment by benleetownsend** [comment]
Are you using multiple GPUs?
effective batch size  = batch_size * num_gpus
Total steps per epoch = dataset size / effective batch size
Total steps  =  num epochs * Total steps per epoch

**Comment by rnyak** [comment]
@benleetownsend I updated my finetune library to the latest version two days ago. For sanity check, I loaded my fine-tuned and saved model from previous model. I get totally different training and test accuracies. In the previous version, my train and test accuracy was 90% and 82%, now with this new release, with the same fine-tuned model, but I am getting 34% for training set, and 16% for test set. That's weird.  

My code lines for fine tuning:
```
import time
start = time.time()
model = Classifier(n_epochs=2 , base_model=GPT2Model, tensorboard_folder ='/workspace/checkpoints', max_length= 1024, val_size = 1000, chunk_long_sequences=False, keep_best_model= True)
model.fit(trainX, trainY)
print("total training time:", time.time() - start)
```
for testing:
```
#Load the saved model
model= Classifier.load('./checkpoints/2epochs_GPT2')
#test accuracy for the test set 
pred_test = model.predict(testX)
accuracy = np.mean(pred_test == testY)
print('Test Accuracy: {:0.3f}'.format(accuracy))
```

**Comment by benleetownsend** [comment]
Is the original issue here resolved?

**Comment by rnyak** [comment]
Yes, we can close the issue.
